astropy/modeling/core.py

```diff
@@ -1551,6 +1551,14 @@ def copy(self):
 
         return copy.deepcopy(self)
 
+    def deepcopy(self):
```

Comment:
As you said, this is identical to `copy()` and is indeed confusing. Is it appropriate to introduce a warning (deprecation?) in `copy()` saying that it will become shallow copy after x releases?